hoot-core-test/src/test/cpp/hoot/core/conflate/network/OsmNetworkExtractorTest.cpp

```diff
@@ -54,7 +54,7 @@ class OsmNetworkExtractorTest : public HootTestFixture
 
     OsmMapReaderFactory::read(map, "test-files/conflate/network/ToyInput.osm");
 
-    ElementCriterionPtr c(new HighwayCriterion());
+    ElementCriterionPtr c(new HighwayCriterion(map));
```

Comment:
The majority of these changes are adding the map to the `HighwayCriterion` like this.  This allows the criterion to check if the way's parent relation is a building relation.  In some cases some buildings get split because they aren't labeled correctly.  This fixes that problem.